horovod/tensorflow/gradient_aggregation.py

```diff
@@ -241,8 +241,17 @@ def apply_gradients(self, apply_grads_closure, optimizer, *args, **kwargs):
         # If optimizer tracks iterations, we increment it on steps where we
         # are not going to call `apply_gradients()`.
         def increment_optimizer_iteration():
-            if hasattr(optimizer, "_iterations") and optimizer._iterations is not None:
-                return optimizer._iterations.assign_add(1).op
+            # (kvignesh1420): Since all `tf.OptimizerV2` instances have the `iterations`
+            # property for modifying the underlying `optimizer._iterations`, it is safe to use
+            # the property instead of the private variable. For instance, the keras
+            # `LossScaleOptimizer` inherits `tf.Optimizer` and exposes the cleaner `iterations`
+            # property instead of the unsafe `_iterations`.
+
+            # NOTE: For users who wrap their optimizers with `LossScaleOptimizer`, the increment op
+            # before this change would have incorrectly updated the underlying optimizer iterations count,
+            # i.e only once every `backward_passes_per_step`.
+            if hasattr(optimizer, "iterations") and optimizer.iterations is not None:
```

Comment:
These comments will be confusing after the change has landed. I think it would be sufficient to have their content in the PR description text on GitHub.

Reply:
removed them in the commit